apps/emqx_dashboard/src/emqx_dashboard_swagger.erl

```diff
@@ -83,13 +84,14 @@
 -type request_meta() :: #{module => module(), path => string(), method => atom()}.
 
 -type filter_result() :: {ok, request()} | {400, 'BAD_REQUEST', binary()}.
--type filter() :: fun((request(), request_meta()) -> filter_result()).
+-type filter() :: fun((request(), request_meta()) -> filter_result()) | undefined.
 
 -type spec_opts() :: #{
     check_schema => boolean() | filter(),
     translate_body => boolean(),
     schema_converter => fun((hocon_schema:schema(), Module :: atom()) -> map()),
-    i18n_lang => atom() | string() | binary()
+    i18n_lang => atom() | string() | binary(),
+    filter => filter()
```

Comment:
Also, I understand that this is preexisting naming, but it's a bit confusing now that it's part of module interface. Have no good ideas how to call it though, _interceptor_ comes into mind, or something along the lines of `before_request` / `on_request`.

Reply:
`filter`s belong to `minirest` domain, and implement the very filter protocol defined by it; since we do not add any additional abstraction over it maybe it is better to preserve the name for intuitiveness and clarity